src/main/java/com/minecolonies/coremod/commands/AbstractCommandParser.java

```diff
@@ -0,0 +1,675 @@
+            final ForgePermissionNodes forgePermissionNode = actionMenu.getForgePermissionNode();
+
+            final EntityPlayer player = (EntityPlayer) sender;
+            if (!permissionsChecker.hasPermission(forgePermissionNode, player))
+            {
+                // TODO: Do something if permission check fails.
+                // But we don't have permissions set up yet.
+            }
+        }
+
+        for (final ActionArgument executionActionArgument : executionActionArgumentList)
+        {
+            if (!actionMenuState.isValueSet(executionActionArgument))
+            {
+                throw new CommandException(getCommandUsage(sender, executionTreeNode));
+            }
+        }
+
+        final Class<? extends IActionCommand> clazz = actionMenu.getActionCommandClass();
+        try
+        {
+            createInstanceAndExecute(server, sender, actionMenuState, clazz);
+        }
+        catch (final InstantiationException | IllegalAccessException e)
+        {
+            final Logger log = LogManager.getLogger();
+            log.error("Unable to instantiate class %s for command %s ", clazz.getName(), actionMenu.getDescription(), e);
+            throw new CommandException("Unable to instantiate class " + clazz.getName() + " for command " + actionMenu.getDescription(), e);
+        }
+    }
+
+    protected void createInstanceAndExecute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final ActionMenuState actionMenuState,
+            @NotNull final Class<? extends IActionCommand> clazz) throws InstantiationException, IllegalAccessException, CommandException
+    {
+        final IActionCommand actionCommand = clazz.newInstance();
+        actionCommand.execute(server, sender, actionMenuState);
+    }
+
+    private static void throwCommandUsageExceptionIfRequiredArgumentsAreNotProvided(@NotNull final TreeNode<IMenu> executionTreeNode, @NotNull final ActionMenu actionMenu,
+            @NotNull final List<ActionArgument> executionActionArgumentList, @NotNull final ActionMenuState actionMenuState, final String badArgument,
+            @NotNull final ICommandSender sender) throws CommandException
+    {
+        final List<ActionArgument> actionArgumentListForActionMenu = actionMenu.getActionArgumentList();
+        for (final ActionArgument actionArgument : actionArgumentListForActionMenu)
+        {
+            if (actionArgument.isRequired())
+            {
+                boolean foundArgument = false;
+                for (final ActionArgument executionActionArgument : executionActionArgumentList)
+                {
+                    if ((null != executionActionArgument) && actionArgument.getName().equals(executionActionArgument.getName()))
+                    {
+                        if (!actionMenuState.isValueSet(executionActionArgument))
+                        {
+                            if (null == badArgument)
+                            {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 5 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)